packages/suite/src/reducers/suite/suiteReducer.ts

```diff
@@ -488,7 +496,10 @@ export const selectFirmwareHashCheckError = (state: AppState) => {
 const selectIsFirmwareHashCheckEnabledAndHardFailed = (state: AppState): boolean => {
     const error = selectFirmwareHashCheckError(state);
 
-    return error !== null ? !isArrayMember(error, softHashCheckErrors) : false;
+    return error !== null
+        ? // broaden the error type, so that `softHashCheckErrors` is assignable as a subset of `error`
+          !isArrayMember(error as FirmwareHashCheckError, softHashCheckErrors)
+        : false;
```

Comment:
This is different for hash check, because `skippedHashCheckErrors`. No such thing yet for revision check.